test/routes.spec.js

```diff
@@ -0,0 +1,297 @@
+        chai.request(server)
+        .delete('/api/v1/favorites/1')
+        .end((err, response) => {
+            response.should.have.status(204);
+          done();
+        });
+    });
+    it('should return a 404 for a failed delete favorite', done => {
+        chai.request(server)
+        .delete('/api/v1/favorites/100')
+        .end((err, response) => {
+            response.should.have.status(404);
+          done();
+        });
+    });
+    it('should return playlists', done => {
+        chai.request(server)
+        .get('/api/v1/playlists')
+        .end((err, response) => {
+            response.should.have.status(200);
+            response.should.be.json;
+            response.body.should.be.a('array');
+            response.body[0].should.have.property('id')
+            response.body[0].should.have.property('name');
+            response.body[0]['favorites'][0].should.have.property('song_title');
+            response.body[0]['favorites'][0].should.have.property('artist_name');
+            response.body[0]['favorites'][0].should.have.property('genre');
+            response.body[0]['favorites'][0].should.have.property('song_rating');
+            done();
+        });
+    });
+    it('should return a playlist', done => {
+        chai.request(server)
+        .get('/api/v1/playlists/1/favorites')
+        .end((err, response) => {
+            response.should.have.status(200);
+            response.should.be.json;
+            response.body.should.be.a('Object');
+            response.body.should.have.property('id');
+            response.body['favorites'][0].should.have.property('song_title');
+            response.body['favorites'][0].should.have.property('artist_name');
+            response.body['favorites'][0].should.have.property('genre');
+            response.body['favorites'][0].should.have.property('song_rating');
+            done();
+        });
+    });
+    it('should return a 404 for a playlist that does not exist', done => {
+        chai.request(server)
+        .get('/api/v1/playlists/100/favorites')
+        .end((err, response) => {
+            response.should.have.status(404);
+            response.should.be.json;
+            response.body.should.be.a('Object');
+            response.body.should.have.property('error')
+            response.body.error.should.equal('Could not find playlist with id 100');
+          done();
+        });
+    });
+    it('should post a favorite to playlist', done => {
+        chai.request(server)
+        .post('/api/v1/playlists/1/favorites/1')
+        .end((err, response) => {
+            response.should.have.status(201);
+            response.should.be.json;
+            response.body.should.be.a('Object');
+            response.body.should.have.property('message');
+            done();
+        });
+    });
+    xit('should 404 when a favorite does not exist when posting to playlist', done => {
```

Comment:
Why are these tests skipped? I'd keep the PR unmerged until they're sorted?

Reply:
On It! Thanks